pkg/kv/bulk/sst_batcher.go

```diff
@@ -241,12 +241,15 @@ func MakeSSTBatcher(
 func MakeStreamSSTBatcher(
 	ctx context.Context,
 	db *kv.DB,
+	rc *rangecache.RangeCache,
 	settings *cluster.Settings,
 	mem mon.BoundAccount,
 	sendLimiter limit.ConcurrentRequestLimiter,
 	onFlush func(summary kvpb.BulkOpSummary),
 ) (*SSTBatcher, error) {
-	b := &SSTBatcher{db: db, settings: settings, ingestAll: true, mem: mem, limiter: sendLimiter}
+	// A mutex is needed because flushes on range boundaries can Grow and Shrink memory asynchronously.
+	mem.Mu = &syncutil.Mutex{}
+	b := &SSTBatcher{db: db, rc: rc, settings: settings, ingestAll: true, mem: mem, limiter: sendLimiter}
```

Comment:
I could see an argument for expecting the caller to pass it in with the correct mutex set and then just checking/asserting that it has a mutex here. But, I'm also OK with it as-is.

Still curious that we don't see this in more places.

Reply:
agreed, and I was thinking even to:
- always have a mutex - if no benchmark will be slower, is it really slower?
- if that's not possible, always have a mutex and add `UnlockedGrow()` and `UnlockedShrink()` (or some better name) to be used when performance is critical, so that it's clear what's going on.

I'll do one of those in a separate pr.